### PR TITLE
CRM-19296 - Improvements in action schedule reminder activity and postemailsend with more params

### PR DESCRIPTION
**Pull request by jitendrapurohit**
issues

- Reminder sent activity does not display actual email content which was sent by scheduled reminder job
- postEmailSend hook does not have enough information if want to do more logging for reminder
- Spool BAO does not return spoolmail object

The above issues are handled in this PR

---

 * [CRM-19296: Reminder Sent acitivity and postEmailSend hook in mailing improvement](https://issues.civicrm.org/jira/browse/CRM-19296)

### CODE REVIEW

**Comment by jitendrapurohit** [comment]
This is a re-submission of https://github.com/civicrm/civicrm-core/pull/8969 with style warning fixes.

File: CRM/Core/BAO/ActionSchedule.php [hunk 1]
```diff
@@ -304,11 +304,11 @@ public static function sendMailings($mappingID, $now) {
             ->context('actionSearchResult', (object) $dao->toArray());
           foreach ($tokenProcessor->evaluate()->getRows() as $tokenRow) {
             if ($actionSchedule->mode == 'SMS' or $actionSchedule->mode == 'User_Preference') {
-              CRM_Utils_Array::extend($errors, self::sendReminderSms($tokenRow, $actionSchedule, $dao->contactID));
+              CRM_Utils_Array::extend($errors, self::sendReminderSms($tokenRow, $actionSchedule, $dao->contactID, $mapping, $dao->entity_id, $dao->reminderID, $dao->case_id));
             }
 
             if ($actionSchedule->mode == 'Email' or $actionSchedule->mode == 'User_Preference') {
-              CRM_Utils_Array::extend($errors, self::sendReminderEmail($tokenRow, $actionSchedule, $dao->contactID));
+              CRM_Utils_Array::extend($errors, self::sendReminderEmail($tokenRow, $actionSchedule, $dao->contactID, $mapping, $dao->entity_id, $dao->reminderID, $dao->case_id));
```
Comment (seamuslee001) [inline CRM/Core/BAO/ActionSchedule.php]: @jitendrapurohit this is failing in the test suite on "Undefined property: CRM_Core_DAO::$case_id"

**Comment by jitendrapurohit** [comment]
@seamuslee001 @Kajakaran @davejenx I've updated the previous code with some changes. Can anyone review this PR.

Thanks.

**Comment by davejenx** [comment]
I have tested this PR against today's dmaster. Re the 3 points listed in CRM-19296:

> 1. Reminder sent activity does not display actual email content which was sent by scheduled reminder job

**Without PR**
Replicated issue: Reminder Sent activity does not substitute tokens, e.g. says "Greetings {contact.display_name}".

**With PR**
Issue fixed: Reminder Sent activity does substitute tokens.

> 2. postEmailSend hook does not have enough information if want to do more logging for reminder

**Without PR**
$params passed to hook_civicrm_postEmailSend() has fields:
groupName, from, toName, toEmail, subject, entity, entity_id, text, html
where text/html have the message plain text/html with tokens substituted.

**With PR**
$params passed to hook_civicrm_postEmailSend() has fields:
groupName, from, toName, toEmail, subject, entity, entity_id, reminder_id, text, html, to, headers, message
where text/html and message have the message plain text/html with tokens substituted.
Of the added fields:
reminder_id: provides new info
to: seems to duplicate to Email
headers: provides new info
message: full message including MIME boundaries, which means it duplicates text+html. Are the MIME boundaries useful or does this provide any other info not already in the text+html fields?

> 3. Spool BAO does not return spoolmail object

Not sure what would be an appropriate test for this.

**Comment by davejenx** [comment]
How does this PR relate to #10081 for CRM-20361 ?

**Comment by eileenmcnaughton** [comment]
@jitendrapurohit can you review the status of this - it's not clear to me if it is just stale or completely superceded

**Comment by eileenmcnaughton** [comment]
@jitendrapurohit I'm going to close this & leave to you to re-open as appropriate as it is stale and it's unclear whether it's still required in it's current state

**Pull request closed without merging**